modules/mergeImageWithTelemetry/mergeImageWithTelemetryWorker.py

```diff
@@ -0,0 +1,45 @@
+from modules.mergeImageWithTelemetry.mergeImageWithTelemetry import MergeImageWithTelemetry
+
+def pipelineMergeWorker(pause, exitRequest, imagePipelineIn, telemetryPipelineIn, pipelineOut):
+    print("start pipeline merge")
+    mergeImageWithTelemetry = MergeImageWithTelemetry()
+    shouldGetImage = True
+    curImage = None
+
+    while True: 
+        if not exitRequest.empty():
+            return
+        
+        pause.acquire()
+        pause.release()
+        
+        if shouldGetImage:
+            curImage = imagePipelineIn.get()
+
+        if curImage is None:
+            continue
+
+        # put current elements of the telemetry pipeline into the queue
+        while not telemetryPipelineIn.empty(): 
```

Comment:
Use locking logic as seen in geolocationWorker, acquire lock, dump queue, release lock

Reply:
I believe this is an attempt to continuously get telemetry data to match to the image. Unfortunately, once the image is matched, it will still continuously get more telemetry data. There is a possibility of spinning forever as the process putting telemetry into the pipeline can do so at an equal or faster rate than this process can extract it.

Additionally, it is relatively expensive to dribble the telemetry into the class object.